code-challenges.js

```diff
@@ -23,19 +56,107 @@ var middleLetters1 = "hello"
+// console.log(middleLetter(middleLetters2));
 
 
 
-// --------------------4) Create a CLASS to get the area of a sphere. Create three spheres with different radi as test cases. Area of a sphere =  4πr^2 (four pi r squared)
 
+// --------------------4) Create a CLASS to get the area of a sphere. Create three spheres with different radi as test cases. Area of a sphere =  4πr^2 (four pi r squared)
 
+//created a class called Sphere with the constructor radius, the value is taken as an argument.
+class Sphere {
+    constructor(radius){
+        this.radius= radius;
+    }
+    //the get syntax binds a property to a function that will be called when that property is looked up.
+    get area() {
+        return this.calcArea();
+    }
+    //this method is called by the get syntax and calculates the area by plugging in the radius into the equation.
+    calcArea() {
+        return 4 * 3.14 * (this.radius * this.radius);
+    }
+}
+
+//here we create three new instances of spheres with the radius plugged in as an argument and in the console log we look up the area of these new spheres, which triggers the get syntax in the class.
+var ball = new Sphere(10);
+// console.log(ball.area);
+//google says expected result ~1256.64, we get 1256
+
+var marble = new Sphere(2);
+// console.log(marble.area);
+//expected ~ 50.26, we get 50.24
+
+var bearing = new Sphere(0.5);
+// console.log(bearing.area);
+//expected ~3.14, we get 3.14
 
 
 // --------------------5) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 var numbersToAdd1 = [2, 4, 45, 9]
-// Excpected output: [2, 6, 51, 60]
+// Expected output: [2, 6, 51, 60]
 var numbersToAdd2 = [0, 7, -8, 12]
 // Expected output: [0, 7, -1, 11]
 var numbersToAdd3 = []
 // Expected output: []
+
+//create a function that takes in an array
+// const addEmAll = (array) => {
+//     //we want a new array to return the added numbers.
+//     let addedArr = []
+//     //loop through every element in the array
+//     //get the sum of an accumulating sum
+//     for (i = 0; i < 2; i++){
+//         //starting value is 0 and add array[i]
+//         let num1= i
+//         let num2= array[i]
+//         let newNum = num1 + num2
+//         // console.log(newNum);
+//         // console.log(num1);
+//         // console.log(num2);
+//         num1 = newNum
+//         console.log(num1);
+//         addedArr.push(newNum)
+//     }
+//     return addedArr
+// }
+
+const addEmAll = (array) => {
+    //create an array the will hold the sums 
+    let sumArr = []
+    for (let i = 0; i < array.length, i++)
+}
+
+//I'm struggling hard with this challenge
+//PSEUDO CODE
+// I am starting at 0
+// I want to add 0 and the first value of the array (2)
+//I want to push that value on to the array of sums
+//take that new value and add the next value of the array (4)
+//push and repeat
+// console.log(addEmAll(numbersToAdd1));
```

Comment:
I think you've got some good ideas here. I'd love to see you finish this up if you have time!